app/models/agents/http_status_agent.rb

```diff
@@ -66,7 +66,8 @@ def check
     def receive(incoming_events)
       incoming_events.each do |event|
         interpolate_with(event) do
-          check_this_url interpolated[:url], header_array(interpolated[:headers_to_save])
+          check_this_url event.payload[:url] || interpolated[:url],
+                         header_array(event.payload[:headers_to_save] || interpolated[:headers_to_save])
```

Comment:
I understand that, but this Agent is relatively young and unmatured, and there are contradictions between the written specs and the implementation.  We have the core contributors/users here, so I take this as a good opportunity to hear from them, know the original intentions and get it right rather than worrying too much about compatibility.